buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/BatsOverVagrantTask.groovy

```diff
@@ -27,12 +27,19 @@ import org.gradle.api.tasks.Input
 public class BatsOverVagrantTask extends VagrantCommandTask {
 
     @Input
-    String command
+    String remoteCommand
 
     BatsOverVagrantTask() {
-        project.afterEvaluate {
-            args 'ssh', boxName, '--command', command
+        command = 'ssh'
+    }
+
+    void setRemoteCommand(String remoteCommand) {
+        if (remoteCommand == null) {
+            // Sanity check, otherwise you might get a confusing NPE when reading args
+            throw new IllegalArgumentException("remoteCommand == null!");
         }
+        this.remoteCommand = remoteCommand
```

Comment:
Just use `this.remoteCommand = Objects.requireNonNull(remoteComment)` instead of the `if` above?